Gems/ROS2/Code/Source/VehicleDynamics/DriveModels/AckermannDriveModel.cpp

```diff
@@ -125,43 +120,19 @@ namespace ROS2::VehicleDynamics
             return;
         }
 
-        const double deltaTimeSec = double(deltaTimeNs) / 1e9;
         for (const auto& wheelData : m_driveWheelsData)
         {
-            auto wheelEntity = wheelData.m_wheelEntity;
-            AZ::Transform wheelTransform;
-            AZ::TransformBus::EventResult(wheelTransform, wheelEntity, &AZ::TransformBus::Events::GetWorldTM);
-
-            AZ::Transform inverseWheelTransform = wheelTransform.GetInverse();
-            AZ::Vector3 currentAngularVelocity;
-            Physics::RigidBodyRequestBus::EventResult(currentAngularVelocity, wheelEntity, &Physics::RigidBodyRequests::GetAngularVelocity);
-            currentAngularVelocity = inverseWheelTransform.TransformVector(currentAngularVelocity);
-            auto currentAngularSpeedX = currentAngularVelocity.Dot(wheelData.m_driveAxis);
+            const auto wheelEntity = wheelData.m_wheelEntity;
             float wheelRadius = wheelData.m_wheelRadius;
-            if (AZ::IsClose(wheelRadius, 0.0f))
-            {
-                const float defaultFloatRadius = 0.35f;
-                AZ_Warning("ApplySpeed", false, "Wheel radius is zero (or too close to zero), resetting to default %f", defaultFloatRadius);
-                wheelRadius = defaultFloatRadius;
-            }
-
-            auto desiredAngularSpeedX = speed / wheelRadius;
-            double pidCommand = m_speedPid.ComputeCommand(desiredAngularSpeedX - currentAngularSpeedX, deltaTimeNs);
-            if (AZ::IsClose(pidCommand, 0.0))
-            {
-                continue;
-            }
-
-            auto impulse = pidCommand * deltaTimeSec;
-
-            auto transformedTorqueVector = wheelTransform.TransformVector(wheelData.m_driveAxis * impulse);
-            Physics::RigidBodyRequestBus::Event(wheelEntity, &Physics::RigidBodyRequests::ApplyAngularImpulse, transformedTorqueVector);
+            const auto hingeComponent = wheelData.m_hingeJoint;
+            const auto id = AZ::EntityComponentIdPair(wheelEntity, hingeComponent);
+            auto desiredAngularSpeedX = (speed / wheelRadius);
```

Comment:
Just to be safe, can we add an assert here to prevent divide by zero?
```suggestion
            AZ_Assert( wheelRadius != 0, "wheelRadius must be non-zero");
            auto desiredAngularSpeedX = (speed / wheelRadius);
```